### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by ethanmusser**
Fixes some bugs in the logging macro formatting, adds some additional logging levels, and provides a verbose logging option to prevent bogging down the system with excessive printing.

### Changelist

- Fixes bugs with the formatting of log messages with newlines
- Fixes incomplete `PRINT_FMT` printing macro for general purpose formatted printing
- Provides additional log severity levels: `TRACE` and `NOTICE`
- Modifies `LOG_*` macros to be non-verbose and adds separate, verbose log macros `VLOG_*`
- Adds additional ANSI graphics and color modes
- Fixes documentation in `log.h` to work with Doxygen

### Testing

Tested in the example files. The default output looks like the following, where the top two blocks are non-verbose, and the bottom two blocks are verbose.

image

### Future Considerations

It might be worth moving to a proper logging library.  [spdlog](https://github.com/gabime/spdlog) looks promising.  [glog](https://github.com/google/glog) is another popular solution.

### CODE REVIEW

File: include/kodlab_mjbots_sdk/log.h [hunk 1]
```diff
@@ -4,199 +4,575 @@
...
- *  - 'DEBUG' (0)
- *  - 'INFO' (1)
- *  - 'WARN' (2)
- *  - 'ERROR' (3)
- *  - 'FATAL' (4)
- *  - 'NONE' (5)
- *  .
- * The minimum level for console output is set by defining `LOG_MIN_SEVERITY`
- * (default is `DEBUG`).
+ * `TRACE`, `DEBUG`, `INFO`, `NOTICE`, `WARN`, `ERROR`, and `FATAL`.
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/log.h:12]: Notice is a better more standard name.

File: include/kodlab_mjbots_sdk/log.h [hunk 2]
```diff
@@ -4,199 +4,575 @@
...
-#define COLOR_GREEN "32"
-#define COLOR_YELLOW "33"
-#define COLOR_BLUE "34"
-#define COLOR_MAGENTA "35"
-#define COLOR_CYAN "36"
-#define COLOR_WHITE "37"
-
-// Color Convenience
+#define STDERR stderr  ///< Standard error stream
+#define STDOUT stdout  ///< Standard output stream
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/log.h:81]: Some of these more common words may want to have some namespace-like terms added.
`STDERR_LOGH` for example.
Not an issue now, but could be if other files ever define similar macros. 
Leave as is for now.

File: include/kodlab_mjbots_sdk/log.h [hunk 3]
```diff
@@ -4,199 +4,575 @@
...
+   * @param message string to be formatted and logged
+   * @param args string format arguments
+   */
+  #define LOG_TRACE(message, args...) LOG_COLOR(message, TAG_TRACE, COLOR_TRACE, ##args)
+
+  /**
+   * @brief Log verbose message to console with `TRACE` severity level
+   * @param message string to be formatted and logged
+   * @param args string format arguments
+   */
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/log.h:297]: Does doxy handle macro functions fine?

**Review by jdcaporale — APPROVED** [review]
LGTM.

File: include/kodlab_mjbots_sdk/log.h (hunk 1 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/log.h:12]: Haha yeah I did some googling to see what was conventional.

File: include/kodlab_mjbots_sdk/log.h (hunk 3 above)
Comment (ethanmusser) [reply include/kodlab_mjbots_sdk/log.h:297]: Yep the Doxygen looks all nice now!

**Pull request merged**